graphql/subcourse/mutations.ts

```diff
@@ -84,11 +85,41 @@ export class MutateSubcourseResolver {
 
     @Mutation((returns) => Boolean)
     @AuthorizedDeferred(Role.ADMIN, Role.OWNER)
-    async subcourseCancel(@Ctx() context: GraphQLContext, @Arg("subcourseId") subcourseId: number) {
+    async subcourseCancel(@Ctx() context: GraphQLContext, @Arg("subcourseId") subcourseId: number): Promise<Boolean> {
         const subcourse = await getSubcourse(subcourseId);
         await hasAccess(context, "Subcourse", subcourse);
         await prisma.subcourse.update({ data: { cancelled: true }, where: { id: subcourse.id }});
-        // TODO send emails
+        const course = await prisma.course.findUnique({ where: { id: subcourse.courseId} });
+        sendSubcourseCancelNotifications(course, subcourse);
+        return true;
+    }
+
+    @Mutation((returns) => GraphQLModel.Lecture)
+    @AuthorizedDeferred(Role.ADMIN, Role.OWNER)
+    async lectureCreate(@Ctx() context: GraphQLContext, @Arg("subcourseId") subcourseId: number, @Arg("lecture") lecture: PublicLectureInput/*, @Arg("studentId", { nullable: true }) studentId?: number*/): Promise<GraphQLModel.Lecture> {
+        const subcourse = await getSubcourse(subcourseId);
+        await hasAccess(context, "Subcourse", subcourse);
+
+        let currentDate = new Date();
+        if (+lecture.start < +currentDate) {
+            throw new Error(`Inputed lecture of subcourse (${subcourseId}) must happen in the future.`);
+        }
+        let isSubcourseInstructor = await prisma.subcourse_instructors_student.count({where: { subcourseId, studentId: lecture.instructorId }}) > 0;
+        if (!isSubcourseInstructor) {
+            throw new Error(`Inputed instructor (id: ${lecture.instructorId}) is not a subcourse instructor`);
+        }
+        const result = await prisma.lecture.create({ data: { ...lecture, subcourseId }});
+        return result;
+    }
+
+    @Mutation((returns) => Boolean)
+    @AuthorizedDeferred(Role.ADMIN, Role.OWNER)
+    async lectureDelete(@Ctx() context: GraphQLContext, @Arg("lectureId") lectureId: number): Promise<Boolean> {
+        const lecture = await getLecture(lectureId);
+        const subcourse = await getSubcourse(lecture.subcourseId);
```

Comment:
I think we should also prevent the deletion of lectures if (a) they are already in the past and (b) if the subcourse is already published (in that case we might have to inform the participants that the lecture was rescheduled, but I guess we can simply forbid this for now and implement once this becomes an issue)